File: 🟡 Medium/921 - Minimum Add to Make Parentheses Valid/_921MinimumAddtoMakeParenthesesValid.py
class Solution:
    def minAddToMakeValid(self, s: str) -> int:
        stack = []

        for i in range(0, len(s)):
            if len(stack) != 0 and stack[-1] == "(" and s[i] == ")":
                stack.pop()
            else:
                stack.append(s[i])

        return len(stack)

File: 🟡 Medium/921 - Minimum Add to Make Parentheses Valid/_921MinimumAddtoMakeParenthesesValid.cpp
class Solution {
public:
    int minAddToMakeValid(string s) {
        stack<char> stack;

        for (int i = 0; i < s.size(); i++) {
            if (!stack.empty() && stack.top() == '(' && s.at(i) == ')') {
                stack.pop();
            } else {
                stack.push(s.at(i));
            }
        }

        return stack.size();
    }
};